guide/src/examine_error_details.md

```diff
@@ -0,0 +1,176 @@
+            status: Some(
+                "INVALID_ARGUMENT",
+            ),
+            details: [
+                BadRequest(
+                    BadRequest {
+                        field_violations: [
+                            FieldViolation {
+                                field: "document.content",
+                                description: "Must have some text content to annotate.",
+                                reason: "",
+                                localized_message: None,
+                            },
+                        ],
+                    },
+                ),
+            ],
+        },
+        http_status_code: Some(
+            400,
+        ),
+        headers: Some(
+            {
+                "accept-ranges": "none",
+                "x-xss-protection": "0",
+                "x-frame-options": "SAMEORIGIN",
+                "date": "Tue, 01 Apr 2025 22:27:53 GMT",
+                "transfer-encoding": "chunked",
+                "content-type": "application/json; charset=UTF-8",
+                "x-content-type-options": "nosniff",
+                "alt-svc": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000",
+                "vary": "Origin,Accept-Encoding",
+                "server": "scaffolding on HTTPServer2",
+            },
+        ),
+    },
+}
+```
+
+### Programmatically examining the error details
+
+Sometimes you may need to examine the error details programmatically. In the
+rest of the example we will traverse the data structure and print the most
+relevant fields.
+
+Only errors returned by the service contain detailed information, so we first
+query the error to see if it contains the correct error type. If it does, we
+can break down some top-level information about the error:
+
+```rust,ignore
+{{#include ../samples/src/examine_error_details.rs:examine-error-details-service-error}}
+```
+
+And then iterate over all the details:
+
+```rust,ignore
+{{#include ../samples/src/examine_error_details.rs:examine-error-details-service-error}}
+```
+
+The client libraries return a the [`StatusDetails`] enum with the different
+types of error details. In this example we will only examine `BadRequest` errors:
+
+```rust,ignore
+{{#include ../samples/src/examine_error_details.rs:examine-error-details-bad-request}}
+```
+
+A `BadRequest` contains a list of fields that are in violation, we can iterate
+and print the details for each:
+
+```rust,ignore
+{{#include ../samples/src/examine_error_details.rs:examine-error-details-each-field}}
+```
+
+Such information can be useful during development. Other branches of
+`StatusDetails`, such as [`QuotaFailure`] may be useful at runtime to throttle
+an application.
+
+### Expected Output
+
+Typically the output from the error details will look like so:
+
+```text
+  status.code=400, status.message=One of content, or gcs_content_uri must be set., status.status=Some("INVALID_ARGUMENT")
```

Comment:
comment: if we normalize the status codes, we will need to update this text

Reply:
Ack.